examples/opt_serving/generator.py

```diff
@@ -137,6 +137,50 @@ def load_model(self):
             self.num_gpus = alpa.get_global_cluster().num_devices
         logger.info(f"Loading model time: {load_time:.2f}")
 
+    # FIXME(Hao): the current batch uses the topp, reponse_length of the first sentence.
+    def pad_batch(self, inputs, pad_value=None):
+        if pad_value is None:
+            pad_value = self.pad
+        new_inputs = inputs
+        src_lens = [len(input) for input in inputs]
+        max_len = max(src_lens)
+        bs = len(inputs)
+
+        for new_input in new_inputs:
+            ori_len = len(new_input)
+            if len(new_input) < max_len:
+                new_input.extend([pad_value for _ in range(max_len - ori_len)])
+        # pad to bs = MAX_BS
+        if bs < MAX_BS:
+            new_inputs.extend([[pad_value for _ in range(max_len)] for _ in range(MAX_BS - bs)])
+        return new_inputs
+    
+    def __call__(
+        self,
+        inputs,
+        cache_ids,
+        pasts=None,
+    ):
+        assert len(inputs) == 1, "enforcing batch size 1 for now to avoid messy logic with caching past key values"
```

Comment:
I plan to call it frequently, but only one request (batch size 1) at a time since the requests are sequential in nature, so I don't think the batching would make things faster for my use case. So I didn't implement it for now. Is that fine for the current PR?

Reply:
If you only submit one request at a time, the speed will be very slow!
1. You have to pay the network overhead for each request. But with batching, you can amortize the overhead
2. The latency of batch-size =1 and batch-size = 16 is the same. By batching, you can get 16x better throughput.

It is fine for the current PR because it is a separate endpoint. Later, if you find the generation speed too slow, you may apply batching